tests/unittests.py

```diff
@@ -0,0 +1,686 @@
+    'yolov7': 'https://github.com/WongKinYiu/yolov7/releases/download/v0.1/yolov7.pt',
+    'yolov7x': 'https://github.com/WongKinYiu/yolov7/releases/download/v0.1/yolov7x.pt',
+}
+
+
+class ToolCases(unittest.TestCase):
+    V5_SOURCE_FOLDER: str = './weights/'
+    V6R1_SOURCE_FOLDER: str = './weights/'
+    V6R2_SOURCE_FOLDER: str = './weights/'
+    V6R21_SOURCE_FOLDER: str = './weights/'
+    V6R3_SOURCE_FOLDER: str = './weights/'
+    V6R4_SOURCE_FOLDER: str = './weights/'
+    V7_SOURCE_FOLDER: str = './weights/'
+    V8_SOURCE_FOLDER: str = './weights/'
+    URL: str = DEFAULT_URL
+    DOWNLOAD_WEIGHTS: bool = True
+    DELETE_OUTPUT: bool = True
+    
+
+    def convert_yolo(self, file_path: str, shape: Union[int, Tuple[int, int]]=416, version: Literal["v5"] = "v5", 
+                     url: str=URL, file_name:str=OUTPUT_FILE_NAME, log: bool=False, n_shaves=DEFAULT_NSHAVES, use_legacy=DEFAULT_USE_LEGACY_FRONTEND, use_rvc2=DEFAULT_USE_RVC2):
+        """ Uploads Yolo weights and receives zip with compiled blob.
+        
+        :param file_path: Path to .pt weights
+        :param shape: Integer or tuple with width and height - must be divisible by 32
+        :param version: Version of the Yolo model
+        :returns: Path to downloaded zip file
+        :param file_path: Path to .pt weights
+        """
+        # Variables
+        with open(file_path,'rb') as input_file:
+            files = {'file': input_file}
+            values = {
+                'inputshape': shape if isinstance(shape, int) else " ".join(map(str,shape)), 
+                'version': version, 
+                'id': uuid4(),
+                'nShaves': n_shaves,
+                'useLegacyFrontend': use_legacy,
+                "useRVC2": use_rvc2
+            }
+            url = f"{url}/upload"
+            if log:
+                print(url)
+            
+            # Upload files
+            session = requests.Session()
+            try: 
+                with session.post(url, files=files, data=values, stream=True) as r:
+                    r.raise_for_status()
+                    if log:
+                        print(f"Conversion complete. Downloading...")
+                    
+                    with open(file_name, 'wb') as f:
+                        for chunk in r.iter_content(chunk_size=8192): 
+                            # If you have chunk encoded response uncomment if
+                            # and set chunk_size parameter to None.
+                            #if chunk: 
+                            f.write(chunk)
+                    if log:
+                        print(r.status_code)
+                session.close()
+                return r.status_code, file_name
+            except Exception as e:
+                print(e)
+                session.close()
+                return 500, ""
+
+    def test_yolov3tinyu(self):
+        print('Running test_yolov3tinyu...')
```

Comment:
This should be turned into a template. For example:
`_test_yolo`which accepts weights path and model version. Seems it's the same code for most (all?) cases, so would be cleaner.